test/unit/test-postoperator.cpp

```diff
@@ -482,4 +483,143 @@ TEST_CASE("GridFunction export", "[gridfunction][Serial][Parallel]")
     check_files("eigenmode", 1, post_op.GetPadDigitsDefault(),
                 {"E_real", "E_imag", "B_real", "B_imag", "S", "U_e", "U_m"});
   }
+}
+
+TEST_CASE("Dimensional field output", "[postoperator][Serial][Parallel]")
+{
+  // Create iodata.
+  Units units(0.496, 1.453);
+  IoData iodata = IoData(units);
+  iodata.domains.materials.emplace_back().attributes = {1};
+
+  // Create serial mesh.
+  int resolution = 3;
+  double length = 12.2;
+  std::unique_ptr<mfem::Mesh> serial_mesh = std::make_unique<mfem::Mesh>(
+      mfem::Mesh::MakeCartesian3D(resolution, resolution, resolution,
+                                  mfem::Element::TETRAHEDRON, length, length, length));
+
+  // Create parallel mesh.
+  auto comm = Mpi::World();
+  int size = Mpi::Size(comm);
+  auto par_mesh = std::make_unique<mfem::ParMesh>(comm, *serial_mesh);
+  iodata.NondimensionalizeInputs(*par_mesh);
+  Mesh mesh(std::move(par_mesh));
+
+  // Create gridfunctions.
+  int order = 1;
+  mfem::H1_FECollection h1_fec(order, mesh.Dimension());
+  mfem::ParFiniteElementSpace h1_fespace(&mesh.Get(), &h1_fec);
+  mfem::ParGridFunction gridfunc_h1(&h1_fespace);
+
+  mfem::ND_FECollection nd_fec(order, mesh.Dimension());
+  mfem::ParFiniteElementSpace nd_fespace(&mesh.Get(), &nd_fec);
+  mfem::ParGridFunction gridfunc_nd(&nd_fespace);
+
+  mfem::RT_FECollection rt_fec(order, mesh.Dimension());
+  mfem::ParFiniteElementSpace rt_fespace(&mesh.Get(), &rt_fec);
+  mfem::ParGridFunction gridfunc_rt(&rt_fespace);
+
+  std::unique_ptr<GridFunction> E, B, V, A;
+  E = std::make_unique<GridFunction>(nd_fespace, true);
+  B = std::make_unique<GridFunction>(rt_fespace, true);
+  A = std::make_unique<GridFunction>(nd_fespace);
+  V = std::make_unique<GridFunction>(h1_fespace);
+
+  // Create fields, initialize them, and set the corresponding gridfunctions.
+  ComplexVector e(nd_fespace.GetTrueVSize()), b(rt_fespace.GetTrueVSize());
+  Vector a(rt_fespace.GetTrueVSize()), v(h1_fespace.GetTrueVSize());
+  e = std::complex<double>(1.0, 1.0);
+  b = std::complex<double>(1.0, 1.0);
+  a = 1.0;
+  v = 1.0;
+
+  E->Real().SetFromTrueDofs(e.Real());
+  E->Imag().SetFromTrueDofs(e.Imag());
+  B->Real().SetFromTrueDofs(b.Real());
+  B->Imag().SetFromTrueDofs(b.Imag());
+  A->Real().SetFromTrueDofs(a);
+  V->Real().SetFromTrueDofs(v);
+
+  // Create some coefficients.
+  MaterialOperator mat_op(iodata, mesh);
+  std::unique_ptr<mfem::VectorCoefficient> S;
+  std::unique_ptr<mfem::Coefficient> U_e, U_m;
+
+  const double scaling = 1.0;
+  U_e = std::make_unique<EnergyDensityCoefficient<EnergyDensityType::ELECTRIC>>(*E, mat_op,
+                                                                                scaling);
+  U_m = std::make_unique<EnergyDensityCoefficient<EnergyDensityType::MAGNETIC>>(*B, mat_op,
+                                                                                scaling);
+  S = std::make_unique<PoyntingVectorCoefficient>(*E, *B, mat_op, scaling);
```

Comment:
`scaling = 1.0` is default, which leads to cleaner code
```suggestion
  U_e = std::make_unique<EnergyDensityCoefficient<EnergyDensityType::ELECTRIC>>(*E, mat_op);
  U_m = std::make_unique<EnergyDensityCoefficient<EnergyDensityType::MAGNETIC>>(*B, mat_op);
  S = std::make_unique<PoyntingVectorCoefficient>(*E, *B, mat_op);
```